cirq-core/cirq/sim/simulator.py

```diff
@@ -858,6 +858,22 @@ def _final_simulator_state(self):
             self._final_simulator_state_cache = self._final_step_result._simulator_state()
         return self._final_simulator_state_cache
 
+    @property
+    def _substates(self) -> Optional[Sequence['cirq.ActOnArgs']]:
```

Comment:
Is it worth creating a generic subclass of this (`TrialResult`) so that we can return the correct `ActOnArgs` type and avoid using `cast` in the classes that call this function?

Also, do you think this function should be public? Currently there's no public way to get a substate from a trial result.

Reply:
I went ahead and did it since we already do the same with StepResult.